catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/types/AssociationsAttributes.java

```diff
@@ -47,7 +48,7 @@
 
     @Override
     public Set<AttributeDescriptor> getAttributeDescriptors() {
-        return DESCRIPTORS;
+        return Collections.unmodifiableSet(DESCRIPTORS);
```

Comment:
Instead of creating an unmodifiable set on every `getAttributeDescriptors` call, you should create that unmodifiable set in the static block.

```
    private static final Set<AttributeDescriptor> DESCRIPTORS;

    static {
        HashSet<AttributeDescriptor> descriptors = new HashSet<>();
        descriptors.add(new AttributeDescriptorImpl(DERIVED,
                true /* indexed */,
                true /* stored */,
                true /* tokenized */,
                true /* multivalued */,
                BasicTypes.STRING_TYPE));
        descriptors.add(new AttributeDescriptorImpl(RELATED,
                true /* indexed */,
                true /* stored */,
                true /* tokenized */,
                true /* multivalued */,
                BasicTypes.STRING_TYPE));
        DESCRIPTORS = Collections.unmodifiableSet(descriptors);
    }
```